gpt4all-backend/llamamodel.cpp

```diff
@@ -332,6 +367,9 @@ bool LLamaModel::loadModel(const std::string &modelPath, int n_ctx, int ngl)
     }
 #endif
 
+    m_supportsEmbedding = isEmbedding;
```

Comment:
Don't we just need a single state for this then?

Reply:
Right now there are three possible states - both are false (nothing loaded yet), supportsEmbedding is true (embedding model loaded), and supportsCompletion is true (regular LLM loaded). That seemed like the cleaner API to me, because otherwise `supportsCompletion()` would be true on an embedding model until you call loadModel, which was not the case before.